os_dep.c

```diff
@@ -2058,28 +2074,87 @@ void GC_register_data_segments(void)
       return(result);
     }
 # else  /* !defined(__CHERI_PURE_CAPABILITY__) */
+    /* The CheriBSD LLVM compiler declares `etext`, `edata` and `end` as */
+    /* typeless variables. If bdwgc is statically linked with the main   */
+    /* executable, these capabilities are compiled with read-only        */
+    /* permissions and bounds that span the .data and .bss sections.     */
+    /*                                                                   */
+    /* If bdwgc is compiled as a shared library, these symbols are       */
+    /* compiled with zero bounds and cannot be dereferenced. Instead,    */
+    /* read-only capability returned by the loader is utilised           */
+    struct scan_bounds {
+        size_t start;
```

Comment:
Are these definitely `size_t`s and not addresses or similar?

Reply:
I don't need to dereference the pointers. Just the address is needed. To make it more explicit, I could change it to `vaddr_t`. Same result I think. Would that be better ?

Reply:
Yes, `size_t` isn't guaranteed to be able to fit pointer-y thingies in it (`size_t` in purecap mode will be 64 bits), so `vaddr_t` sounds more appropriate to me.

Reply:
changed to `vaddr_t` in  [07b586b](https://github.com/capablevms/bdwgc/pull/19/commits/07b586b7ec011b520062977bd8e863a51061e711)